src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -5,114 +5,283 @@
+        @PreAuthorize("hasAuthority('SCOPE_ROLE_STUDENT')")
+        public List<GradeDTO> getAssignmentGrades ( @PathVariable("assignmentId") int assignmentId,
+                                                    Principal principal){
 
+//        int sectionNo = assignmentRepository.findSectionNoByAssignmentId(assignmentId);
+            Assignment a = assignmentRepository.findById(assignmentId).orElse(null);
+            if(a==null){
+                throw  new ResponseStatusException( HttpStatus.NOT_FOUND, "not found");
+            }
 
+            List<Enrollment> enrollments = enrollmentRepository.findEnrollmentsBySectionNoOrderByStudentName(a.getSection().getSectionNo());
+            List<GradeDTO> assignmentGrades = new ArrayList<>();
 
-    // student lists their assignments/grades for an enrollment ordered by due date
-    // student must be enrolled in the section
-    @GetMapping("/assignments")
-    public List<AssignmentStudentDTO> getStudentAssignments(
-            @RequestParam("studentId") int studentId,
-            @RequestParam("year") int year,
-            @RequestParam("semester") String semester) {
+            for (Enrollment enrollment : enrollments) {
+                Grade grade = gradeRepository.findByEnrollmentIdAndAssignmentId(enrollment.getEnrollmentId(), assignmentId);
 
-        // TODO remove the following line when done
+                if (grade == null) {
+                   grade = new Grade();
+                   grade.setEnrollment(enrollment);
+                   grade.setAssignment(a);
+                   grade.setScore(null);
+                   gradeRepository.save(grade);
+                }
+                    GradeDTO gradeDTO = new GradeDTO(
+                        grade.getGradeId(),
+                        enrollment.getStudent().getName(),
+                        enrollment.getStudent().getEmail(),
+                        grade.getAssignment().getTitle(),
+                        grade.getAssignment().getSection().getCourse().getCourseId(),
+                        grade.getAssignment().getSection().getSecId(),
+                        grade.getScore());
+                    assignmentGrades.add(gradeDTO);
+            }
+            return assignmentGrades;
+        }
 
-        // return a list of assignments and (if they exist) the assignment grade
-        //  for all sections that the student is enrolled for the given year and semester
-		//  hint: use the assignment repository method findByStudentIdAndYearAndSemesterOrderByDueDate
+        // instructor uploads grades for assignment
+        // user must be instructor for the section
+        //TEST URL [{"gradeId":1,"studentName":"thomas edison", "studentEmail":"tedison@csumb.edu","assignmentTitle":"db homework 1","courseId":"cst363","sectionId":1,"score":88}]
 
-        return null;
-    }
-}
+        @PutMapping("/grades")
+        @PreAuthorize("hasAuthority('SCOPE_ROLE_ADMIN')")
+        public void updateGrades (
```

Comment:
Role needs to be changed from 'SCOPE_ROLE_ADMIN' to 'SCOPE_ROLE_INSTRUCTOR'

Reply:
Pushed the update from ADMIN to INSTRUCTOR Thanks @nickleus27